augur/application/db/models/augur_data.py

```diff
@@ -878,7 +878,13 @@ class Repo(Base):
     @staticmethod
     def get_by_id(session, repo_id):
 
-        return session.query(Repo).filter(Repo.repo_id == repo_id).first()
+        try:
+            return session.query(Repo).filter(Repo.repo_id == repo_id).first()
+        except Exception as e:
```

Comment:
Is there a specific exception we can catch? What happens if some other exception occurs which doesn't necessitate a rollback?

Reply:
There wasn't a specific exception that I could find (if you can find one other than rolling back when we get the PendingRollbackException that would be awesome). I don't think calling rollback when nothing happened like during an operational exception should do anything.